pkg/ephemeral/network/tls_connector_test.go

```diff
@@ -0,0 +1,228 @@
+Nqp46LdbAgMBAAECggEBAJ6ViM8AiTn1RmRNImdwSAHLtwZz6ziFtsXUmacGlQRH
+MGLf6WTfCEgkKfd5op7o2Gqc9D8Qk4k+y8hG3jsXZ/owyRcVee0MnRjxbvOA4Q60
+PZFYGjdd5YXX+i2j/T3DOJU4ZcNHPzFLl9kX8Q37z5Nc1TYBXh8sJzW5kCIy5xEL
+XAKNcwGTZF1ml3jkWkFl3LukS3DP8fF1qDvD987YGuc9oVliYW1F0oKL9VGyS7nB
+BtQWslFdP8MbPXG1hjkFydCBiE4teqrFen6hvLdIQk7XJ88Q9UmBoOPJr6+gHuDf
+vk33nVGpBVQ1UHFPnDzZyQKtlDBVEUJ8XhqzEkm4asECgYEA5sTgxJt/nJCL9Lh1
+61jFbVD21SVFEv7IWIV6YjBxzJhzGVJa6ZhrOnRTrkTAkraJ1wUd9FyIdEsL/Nvy
+/z8hOAXbty1zXdpOo/BV0J6zRwJ0Cj8WVTeCUr5KGgw/pzbQdltJ+1J8jnAGbZjN
+Ri/QUdryqZTQz3rD8sDVDFvLojMCgYEAw78HQ+y/gL5Z/IJww0lUYHjqcm1G5taY
+3Ht6qRvkqdCmW8qC2wpKFKl9lCJfo+H1jidjhM5RTPFSlCxiWtxLAamMvfv0f3d6
+q5gPjcjak275bnmU1e0blkLEdeXQljRXH+oDmur95udzh0DrdTDJ/lqbf3uui8Uc
+VApAcSbR/jkCgYAWUT/zg55Jw+jlF9m/kuw08DmOz3Xoql8xwGbfjBPVV4D6F+7W
+3HiyRIG7Psbo6WJXOxV0hmZj6MYWBCdx6+cIhfiDtI+Nqgkk7Z8+97oaye/y9brx
+LtcZrXF5J2oYf8KVT6rN9WI6XDci7j4b5Y/d+rCxGcU/6317wo5YDaCZ5QKBgET3
+4qRxHwxKhUQt5XM5PAx9rgVBMXEV/Wf57b71v/yBMow27yIkHvPmwANYlSAV9kHu
+6OabFxQoFvN0K/ddlOPyDE/IHV5oB4W8HwbS1QiLWkEtf15cm5K21afAoFy79lKd
+TkXgNDOOKytlmVCCLzl6TT1+o4JFofSOZCQ6DFUpAoGAAQdaCjX5UCeWb5e/Vbiu
+SQL1RKIHkgm6gj1UjlQ981r6y+hVkBygtIr/eW0wSkFAkUrOdefHNVOQW18ESF06
+YqBL4gD7aEij9kGd0PrievimgcYYaBHOcO1RouQOURTMmWqjIPu1fyWDv+rFk+S5
+2uCuYndpzOgCiEhjDGCuSug=
+-----END PRIVATE KEY-----`
+
+	pemFileServer = `-----BEGIN CERTIFICATE-----
+MIIC+zCCAeOgAwIBAgIUNI9WRun2Y+ICmpzjYRpVcJ/BBE4wDQYJKoZIhvcNAQEL
+BQAwDTELMAkGA1UEAwwCUDAwHhcNMjIwMTI2MTI0NjQ5WhcNMjIwMjI1MTI0NjQ5
+WjANMQswCQYDVQQDDAJQMDCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEB
+ALB0KqYSOIxUE8qbfUgb4j7HCBf4ScSmesGeU5rRaEXNoyvCpemKOAwD50s1HxrX
+OnKGCVYGyqbx2k3u0RXX2mKLM6bFm1l57uHUd2yj/7hTXTo2rNfkcrC6plVI6NU3
+b1IUqrSw2uJwhfUdpsTktjIFAm3MzPEBnByVv57u4YF8ckOw6TeE6GzduQnJXd5F
+rS4cfYG7gI8fov1tyKwUKYeLKRl5lpj4GlGRLRzn6Hu6YuspIZR7fv+9/1Rj2eIs
+NQ2iUd49L+JCeZse/WX+YOXYu83Lk5IJK84TovQYwJQOdiZh/oX330st1CALuBxk
+Kneu6NngxhSypHY2qnjot1sCAwEAAaNTMFEwHQYDVR0OBBYEFDjtm5a7RbAFeYuQ
+QfFYci+eTOeXMB8GA1UdIwQYMBaAFDjtm5a7RbAFeYuQQfFYci+eTOeXMA8GA1Ud
+EwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggEBAGo1n03gEYMsBLLaOcY7dDwn
+behhLE7UP3eWRw2gpmbKfilk+dljYWsOdiQeXktE/LxyFiuBNwefI7JrypFifzio
+udqYyQAJ2pvMogij+TPajaDhJxmMWqRizcAo/6cXekSCufnRbbTBENUG2ZNHRuyn
+zsYFZtpxDO9LF0uutE2P6NJQpKKrCo/NGMV4AF0vy1tKp6h2fBU3K9Yn+1RihIyS
+Y+sLoNiorJloqZ8qn2cULbax/xi/IcccdRJfoIjmIuSl9wUwl+lkeGB9Rlwm5iFJ
+LO+mQ15hUEpbjrXF3IdY+4MjDqFOETC0KuI72yjUGPZqWe+WAhBcni3VNzs2Ik4=
+-----END CERTIFICATE-----`
+)
+
+var _ = Describe("TlsConnector", func() {
+	var testDataFolder string
+	var certificateFolder string
+	var playerID = int32(0)
+
+	BeforeEach(func() {
+		var err error
+		testDataFolder, err = ioutil.TempDir("", "testData")
+		certificateFolder = testDataFolder + "/Player-Data"
+		err = os.Mkdir(certificateFolder, os.ModeDir|os.ModePerm)
+		if err != nil {
+			panic(err)
+		}
+
+		err = ioutil.WriteFile(fmt.Sprintf("%s/C%d.pem", certificateFolder, playerID), []byte(pemFileClient), os.ModePerm)
```

Comment:
Is there a reason to write/create the cert files as part of the test instead of having them as part of test resources?
Q: What's best practice (for go)?